base/ltmath.dtx

```diff
@@ -370,9 +370,39 @@
 %    \begin{macrocode}
 \def\makesm@sh#1{%
   \setbox\z@\hbox{\color@begingroup#1\color@endgroup}\finsm@sh}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2022/11/01}%
+%<latexrelease>                 {\mathsm@sh}{Guard against reboxing}%
 \def\mathsm@sh#1#2{%
-  \setbox\z@\hbox{$\m@th#1{#2}$}\finsm@sh}
+  \setbox\z@\hbox{$\m@th#1{#2}$}%
+%    \end{macrocode}
+%    The empty brace groups in front of the smashed box (which is
+%    placed by \cs{finsm@sh}) ensures that a \cs{smash} in math is not
+%    just producing a single box  with its dimensions altered, but a
+%    box plus this second ord atom. The reason is that \TeX{} sometimes reboxes a
+%    box if its the only thing in a place like the denominator of a
+%    fraction. This would then undo the smashing and the addiitonal
+%    kern prevents that. Two ord atoms in a row do not alter the
```

Comment:
kern or atom?

Reply:
Also, wrongly placed i in "addiitonal" the line above.

Reply:
yep, atom ... was initially a kern then changed